### PR TITLE
Bottom Panel Implementation

### PR DESCRIPTION
**Pull request by richagadgil**
Implemented a bottom panel to go alongside the top, main, and side panels. Can be populated with widgets. Meant to be groundwork for status bar, etc.
@ellisonbg

### CODE REVIEW

**Comment by ian-r-rose** [comment]
This looks great to me. My only concern is with the height of the bottom panel: it is zero when there is nothing in it (good!), but when something is added to it it remains zero (bad!). Am I missing something? @jasongrout do you think this would be a good candidate for your `ToolbarLayout`?

**Comment by jasongrout** [comment]
Yes, I think it should probably use the toolbar layout, or similar logic.

**Comment by jasongrout** [comment]
Wait. The status bar added to the bottom panel should have the toolbar layout logic. The bottom panel itself should not.

**Comment by richagadgil** [comment]
I just modified the code so that the bottom panel keeps its minimum height, but stays hidden until widgets are added to it. Hope this fixes the problem.

**Comment by ian-r-rose** [comment]
@jasongrout I'm not sure what you mean when you say the bottom panel should not have toolbar layout logic: why not?

**Comment by jasongrout** [comment]
> @jasongrout I'm not sure what you mean when you say the bottom panel should not have toolbar layout logic: why not?

I think for now, the bottom panel should have exactly the same logic as the top panel. Glancing at the code, it looks like it does, so I think we're good!

**Comment by ian-r-rose** [comment]
Well, it's not exactly the same: the top area has a hard-coded height, whereas the bottom area dynamically hides/shows depending on whether there is anything in it. I think that's the right behavior, but I wonder whether we should use this implementation or the `ToolbarLayout` for that.

**Comment by jasongrout** [comment]
> @jasongrout I'm not sure what you mean when you say the bottom panel should not have toolbar layout logic: why not?

The thought process was that if the status bar doesn't have any items in it, then it should auto-collapse its height, and the bottom panel would just follow along, i.e., the bottom panel's min-height would be the min-height of the status bar. However, it's a moot point if the status bar is always displaying no matter if there are widgets or not.

For now, I think it's fine to just have the same logic as the top bar.

**Comment by jasongrout** [comment]
> Well, it's not exactly the same: the top area has a hard-coded height, whereas the bottom area dynamically hides/shows depending on whether there is anything in it.

Do you mean the css min-height is forcing the top and bottom panel to show, even if there is nothing in them, right now?

**Comment by ian-r-rose** [comment]
No, I am referring to https://github.com/jupyterlab/jupyterlab/pull/4752/files#diff-62da0550d0f6d9bceb4bb477732a8e38R470

**Comment by jasongrout** [comment]
(Update: in an in-person video conference, we decided it was better to switch both the top and bottom panels to use a box layout rather than just a panel layout).

File: packages/application/src/shell.ts [hunk 1]
```diff
@@ -96,6 +99,7 @@ class ApplicationShell extends Widget {
     rightHandler.sideBar.addClass('jp-mod-right');
     rightHandler.stackedPanel.id = 'jp-right-stack';
 
+    bottomPanel.direction = 'left-to-right';
```
Comment (ian-r-rose) [inline packages/application/src/shell.ts]: Why change the layout direction to `left-to-right`? I had envisioned people possibly stacking things on top of the status bar.

**Comment by richagadgil** [comment]
@ian-r-rose That makes sense for the outermost components -- changed the direction in the latest commit to 'bottom-to-top.'

**Comment by ian-r-rose** [comment]
Great, that makes sense to me.

**Pull request merged**